### PR TITLE
Create dml-join.md

### PR DESCRIPTION
**Pull request by soyeric128**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

added doc for JOINs

## Changelog

- Documentation

## Related Issues

Fixes #issue

### CODE REVIEW

**Comment by BohuTANG** [comment]
join is a more complex operation. We'd better find a more realistic example to demonstrate them.

File: docs/doc/30-reference/30-sql/20-query-syntax/dml-join.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
+---
+title: JOIN
+---
+
+A *join* allows you to combine columns from two or more tables into a single result set. Databend supports the following *join* types:
+
```
Comment (xudong963) [inline docs/doc/30-reference/30-sql/20-query-syntax/dml-join.md:6]: Better to have a general structure, like:
```sql
SELECT <expr_list>
FROM <left_table>
[GLOBAL] [INNER|LEFT|RIGHT|FULL|CROSS] [OUTER|SEMI|ANTI|ANY|ASOF] JOIN <right_table>
(ON <expr_list>)|(USING <column_list>) ...
```
From clickhouse doc

**Comment by xudong963** [comment]
> join is a more complex operation. We'd better find a more realistic example to demonstrate them.

I agree. realistic table + realistic query. Do you have a dataset you prefer?

**Comment by PsiACE** [comment]
> I agree. realistic table + realistic query. Do you have a dataset you prefer?

I think it's fine to keep using ontime?

File: docs/doc/30-reference/30-sql/20-query-syntax/dml-join.md (hunk 1 above)
Comment (soyeric128) [reply docs/doc/30-reference/30-sql/20-query-syntax/dml-join.md:6]: Can we have this structure until when we support them all ?

**Comment by BohuTANG** [comment]
> > join is a more complex operation. We'd better find a more realistic example to demonstrate them.
> 
> I agree. realistic table + realistic query. Do you have a dataset you prefer?

`Orders` and `Customers`  is ok, there are many cases:
https://www.w3schools.com/sql/sql_join.asp#:~:text=Different%20Types%20of%20SQL%20JOINs&text=(INNER)%20JOIN%20%3A%20Returns%20records,records%20from%20the%20left%20table

**Comment by BohuTANG** [comment]
> > I agree. realistic table + realistic query. Do you have a dataset you prefer?
> 
> I think it's fine to keep using ontime?

Please don't use that :D

**Comment by soyeric128** [comment]
Examples improved in the latest commit.

**Review by xudong963 — APPROVED** [review]
Would you like to take another look?

**Pull request merged**